model/model_training/trainer_sft.py

```diff
@@ -186,6 +188,12 @@ def argument_parsing(notebook=False, notebook_args=None):
     conf["local_rank"] = args.local_rank
     conf["deepspeed"] = args.deepspeed
 
+    # get the world size in deeepspeed
+    if conf["deepspeed"]:
+        conf["world_size"] = int(os.environ["WORLD_SIZE"])
```

Comment:
Should it fail if not set or could we use `int(os.getenv("WORLD_SIZE", default="1"))`?

Reply:
Should always work with deepspeed but sure better to be safe